plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java

```diff
@@ -120,7 +149,13 @@ public Page getNextPage()
                     blocks[i] = dataPage.getBlock(delegateIndexes[i]);
                 }
             }
-            return new Page(batchSize, blocks);
+
+            CloseableIterable<TrinoRow> filteredRows = deleteFilter.filter(CloseableIterable.withNoopClose(TrinoRow.fromBlocks(columnTypes, blocks, batchSize)));
+            int[] positionsToKeep = StreamSupport.stream(filteredRows.spliterator(), false)
```

Comment:
`[Streams.]stream(filteredRows)`

however, we avoid using streams in "execution" code (eg invoked per row or per page), for their slight overhead.